src/Features/Core/Portable/NavigateTo/AbstractNavigateToSearchService.InProcess.cs

```diff
@@ -187,44 +199,19 @@ private static string GetItemKind(DeclaredSymbolInfo declaredSymbolInfo)
 
         private static NavigateToMatchKind GetNavigateToMatchKind(ArrayBuilder<PatternMatch> nameMatches)
         {
-            if (nameMatches.Any(r => r.Kind == PatternMatchKind.Exact))
-            {
-                return NavigateToMatchKind.Exact;
-            }
-
-            if (nameMatches.Any(r => r.Kind == PatternMatchKind.Prefix))
-            {
-                return NavigateToMatchKind.Prefix;
-            }
-
-            if (nameMatches.Any(r => r.Kind == PatternMatchKind.Substring))
-            {
-                return NavigateToMatchKind.Substring;
-            }
-
-            if (nameMatches.Any(r => r.Kind == PatternMatchKind.CamelCaseExact))
-            {
-                return NavigateToMatchKind.CamelCaseExact;
-            }
+            // work backwards through the match kinds.  That way our result is as bad as our worst match part.  For
+            // example, say the user searches for `Console.Write` and we find `Console.Write` (exact, exact), and
+            // `Console.WriteLine` (exact, prefix).  We don't want the latter hit to be considered an `exact` match, and
+            // thus as good as `Console.Write`.
```

Comment:
💭 I'll be interested to see if this has any negative side effects, but don't see the harm in trying the approach.